ENVParser/EnvFile.cs

```diff
@@ -0,0 +1,580 @@
+            Field23C = reader.ReadSingle();
+            GlareLength = reader.ReadSingle();
+            GlareChromaticAberration = reader.ReadSingle();
+            GlareDirection = reader.ReadSingle();
+            GlareMode = reader.ReadUInt32();
+            Field250 = reader.ReadBoolean();
+            Field251 = reader.ReadUInt32();
+            Field255 = reader.ReadSingle();
+            Field259 = reader.ReadSingle();
+            Field25D = reader.ReadSingle();
+            EnableDOF = reader.ReadBoolean();
+            DOF_FocalPlane = reader.ReadSingle();
+            DOF_NearBlurPlane = reader.ReadSingle();
+            DOF_FarBlurPlane = reader.ReadSingle();
+            DOF_FarBlurLimit = reader.ReadSingle();
+            DOF_BlurScale = reader.ReadSingle();
+            DOF_GaussType = reader.ReadUInt32();
+            EnableSSAO = reader.ReadBoolean();
+            SSAO_OccluderRadius = reader.ReadSingle();
+            SSAO_FallOffRadius = reader.ReadSingle();
+            SSAO_BlurScale = reader.ReadSingle();
+            SSAO_Brightness = reader.ReadSingle();
+            SSAO_DepthRange = reader.ReadSingle();
+            DisableUnknownFlaggedSection = reader.ReadBoolean();
+            FieldShadowFarClip = reader.ReadSingle();
+            Field294 = reader.ReadSingle();
+            AmbientShadowBrightness = reader.ReadSingle();
+            Field29C = reader.ReadSingle();
+            Field2A0 = reader.ReadUInt32();
+            FieldShadowNearClip = reader.ReadSingle();
+            FieldShadowBrightness = reader.ReadSingle();
+            Field2AC = reader.ReadBoolean();
+            Field2AD = reader.ReadBoolean();
+            Field2AE = reader.ReadBoolean();
+            Field2AF = reader.ReadBoolean();
+            ShadowColourRed = reader.ReadSingle();
+            ShadowColourGreen = reader.ReadSingle();
+            ShadowColourBlue = reader.ReadSingle();
+            ShadowColourAlpha = reader.ReadSingle();
+            DisplayColourGrading = reader.ReadBoolean();
+            Cyan = reader.ReadSingle();
+            Magenta = reader.ReadSingle();
+            Yellow = reader.ReadSingle();
+            Dodge = reader.ReadSingle();
+            Burn = reader.ReadSingle();
+            LightMapR = reader.ReadSingle();
+            LightMapG = reader.ReadSingle();
+            LightMapB = reader.ReadSingle();
+            LightMapA = reader.ReadSingle();
+            EnableOutline = reader.ReadBoolean();
+            OutlineOpacity = reader.ReadSingle();
+            OutlineWidth = reader.ReadSingle();
+            CharacterOutlineBrightness = reader.ReadSingle();
+            Field2F2 = reader.ReadSingle();
+            Field2F6 = reader.ReadSingle();
+            ReflectionHeight = reader.ReadSingle();
+            EnablePhysicsSection = reader.ReadBoolean();
+            Gravity = reader.ReadSingle();
+            EnableWind = reader.ReadBoolean();
+            WindDirectionX = reader.ReadSingle();
```

Comment:
Could be worth having a position (X, Y, Z) class, too.

Reply:
Hmmm tricky one, because the planes area not clearly defined and they might use different coordinate origins (character vs camera vs something else), I'm a little hesitant on this one.

Reply:
I don't think having different origins is a dealbreaker if they're still ultimately coordinates. That seems like something you could specify in the name once you're certain of it. Or I suppose you could call some of them `Vector` instead of `Coordinates`, if that makes more conceptual sense.